### PR TITLE
replaced mem buffer reallocation with zeroing it

### PR DESCRIPTION
**Pull request by khodzha**
I'm not sure if this is proper way to address #52 but it seems unnecessary to reallocate whole mem vec when you can just set each element to zero inside and reuse it again?

### CODE REVIEW

**Comment by khodzha** [comment]
I removed other vector allocations that seem unnecessary too

File: src/speex_resample.rs [hunk 1]
```diff
@@ -636,9 +636,9 @@ impl SpeexResamplerState {
      */
     pub fn reset_mem(&mut self) {
         let nb_channels = self.nb_channels as usize;
-        self.last_sample = vec![0; nb_channels];
-        self.magic_samples = vec![0; nb_channels];
-        self.samp_frac_num = vec![0; nb_channels];
+        self.last_sample.clear(); self.last_sample.resize(nb_channels, 0);
+        self.magic_samples.clear(); self.magic_samples.resize(nb_channels, 0);
+        self.samp_frac_num.clear(); self.samp_frac_num.resize(nb_channels, 0);
```
Comment (Luni-4) [inline src/speex_resample.rs]: Could you use the same approach of `self.mem`, please? It's much clearer, thanks!

**Comment by Luni-4** [comment]
@khodzha 

Great! Please squash the commits and use a commit message consistent with the ones used on the repo and we are ready to merge in

**Comment by khodzha** [comment]
@Luni-4 is this commit description okay?

**Comment by Luni-4** [comment]
> @Luni-4 is this commit description okay?

Sure! Thanks a lot for your contribution! :)

**Pull request merged**